JM/CF/Scripts/4_World/CommunityFramework/ModStorage/CF_ModStorageObject.c

```diff
@@ -52,27 +67,43 @@ class CF_ModStorageObject<Class T> : CF_ModStorageBase
 
 	override bool OnStoreLoad(ParamsReadContext ctx, int version)
 	{
-#ifdef CF_TRACE_ENABLED
+#ifdef CF_MODSTORAGE_TRACE
 		auto trace = CF_Trace_2(this, "OnStoreLoad").Add(ctx).Add(version);
 #endif
 
 		// Clearing the unloaded mods data
 		m_UnloadedMods.Clear();
 
 		// Persistence version is prior to 1.10
-		if (version < 116)
+		if (version < CF_ModStorage.GAME_VERSION_FIRST_INSTALL)
 		{
 			return true;
 		}
 
+		int b1, b2, b3, b4;
+		m_Entity.GetPersistentID(b1, b2, b3, b4);
+
+		// If the version is less than the wipe file, the entity will be added automatically in 'OnStoreSave'
+		if (version >= CF_ModStorage.GAME_VERSION_WIPE_FILE)
+		{
+			if (!m_Module.IsEntity(b1, b2, b3, b4))
+			{
+				// Since the entity wasn't found we can assume that CF is freshly loaded
+				// Highly unlikely anything else happened that can cause this
+				// * A new entity won't take the ID of an old entity if it is in circulation
+				// * If this is a new entity, OnStoreSave would've been called at some point before 'OnStoreLoad'
+				return true;
+			}
+		}
+
```

Comment:
So essentially we are saying, if the entity is not in the bin file, we do not attempt to read the `cf_version` from its buffer. After the first save of that entity with CF installed it should be in the file, hence has at least the `cf_version` data written to it, so it is safe to do the next step. And once we know the `cf_version` we can know what else we are allowed to read?

And all we need the dayz update for is to say "alright existing installations do not have the file yet, so we need to read the `cf_version` because it should already be in the data" and after that happend one time they have their files and everything is fine. So we use the `version` incrementing as "trigger once" for our upgrade process?

The name `GAME_VERSION_WIPE_FILE` confused me a little. Not sure what this is supposed to mean. Version `127` will be the next major stable update?

Reply:
> So essentially we are saying, if the entity is not in the bin file, we do not attempt to read the cf_version from its buffer. After the first save of that entity with CF installed it should be in the file, hence has at least the cf_version data written to it, so it is safe to do the next step. And once we know the cf_version we can know what else we are allowed to read?

Correct

> And all we need the dayz update for is to say "alright existing installations do not have the file yet, so we need to read the cf_version because it should already be in the data" and after that happend one time they have their files and everything is fine. So we use the version incrementing as "trigger once" for our upgrade process?

Correct

> The name GAME_VERSION_WIPE_FILE confused me a little. Not sure what this is supposed to mean. Version 127 will be the next major stable update?

Correct, storage version 127 corresponds to game version 1.16